plugins/notifications-backend/src/service/handlers.ts

```diff
@@ -97,12 +106,32 @@ export function getNotifications(
         topic: message.topic,
       };
       return notification;
+    }),
+  );
+
+  const actionsMessageIds = notifications.map(notification => notification.id);
+
+  const actionsQuery = dbClient('actions')
+    .select('*')
+    .whereIn('messages_id', actionsMessageIds);
+  await actionsQuery.then(actions => {
+    actions.forEach(action => {
+      const notification = notifications.find(n => n.id === action.messages_id);
+      if (notification) {
+        notification.actions = notification.actions || [];
```

Comment:
i think it would be nicer to put an empty array in the loop over notifications. this way you avoid all these 'if' and '||'

Reply:
ok, so let's make the `Notification.actions` mandatory - we will always return something.

Reply:
done